Source/WebKit/UIProcess/mac/WKSharingServicePickerDelegate.mm

```diff
@@ -151,21 +151,23 @@ - (void)sharingService:(NSSharingService *)sharingService didShareItems:(NSArray
         WeakPtr weakPage = menuProxy ? menuProxy->page() : nullptr;
         RetainPtr<NSString> itemUTI = itemProvider.get().registeredTypeIdentifiers.firstObject;
         [itemProvider loadDataRepresentationForTypeIdentifier:itemUTI.get() completionHandler:[weakPage, attachmentID = _attachmentID, itemUTI](NSData *data, NSError *error) {
-            RefPtr webPage = weakPage.get();
-            
-            if (!webPage)
-                return;
-            
-            if (error)
-                return;
-            
-            auto apiAttachment = webPage->attachmentForIdentifier(attachmentID);
-            if (!apiAttachment)
-                return;
-            
-            RetainPtr attachment = wrapper(apiAttachment);
-            [attachment setData:data newContentType:itemUTI.get()];
-            webPage->didInvalidateDataForAttachment(*apiAttachment.get());
+            ensureOnMainRunLoop([weakPage = WTFMove(weakPage), attachmentID, itemUTI = WTFMove(itemUTI), data = retainPtr(data), error = retainPtr(error)] {
```

Comment:
`itemUTI = WTFMove(itemUTI)` this doesn't actually move, so we can just not move

Reply:
Also, I think our static analysis tooling prefers direct `RetainPtr` construction (rather than using the `retainPtr(...)` helper method.

Reply:
In all...
```suggestion
            ensureOnMainRunLoop([weakPage = WTFMove(weakPage), attachmentID, itemUTI, data = RetainPtr { data }, error = RetainPtr { error }] {
```